// vite-plugin-testing/dist/index.mjs
import image from '@rollup/plugin-image';
import vue from '@vitejs/plugin-vue';
import { resolve } from 'path';
import readline from 'readline';
import { viteStaticCopy } from 'vite-plugin-static-copy';

const defaultUrl = "https://staging.repairdesk.co/u/start-serve";
const defaultItem = {
  src: "src/options.json",
  dest: ""
};
function cyan(str) {
  return `\x1B[36m${str}\x1B[0m`;
}
const handleOpenURL = (openUrl) => ({
  name: "handle-open-url",
  configureServer(server) {
    const onInput = (input) => {
      if (input === "\r" || input === "\n") {
        console.log("Enter key pressed");
        import('open').then((res) => {
          res.default(openUrl);
        });
      }
      if (input === "" || input === "") {
        server.close().finally(() => process.exit(1));
      }
    };
    readline.emitKeypressEvents(process.stdin);
    process.stdin.setRawMode(true);
    process.stdin.on("data", onInput).setEncoding("utf8").resume();
    server.httpServer.on("close", () => {
      process.stdin.off("data", onInput).pause();
    });
    server.httpServer?.on("listening", () => {
      setTimeout(() => {
        setTimeout(() => {
          server.config.logger.info(`
  ${cyan("Press Enter to open url")} `);
        });
      });
    });
  }
});
function index(options, env) {
  const plugins = [
    {
      ...image(),
      enforce: "pre"
    },
    vue(),
    handleOpenURL(defaultUrl)
  ];
  if (env == "build") {
    plugins.push(viteStaticCopy({
      targets: !!options.icon ? [
        defaultItem,
        {
          src: resolve(__dirname, options.icon),
          dest: "icon"
        }
      ] : [defaultItem]
    }));
  }
  return plugins;
}

export { index as default };
